lib/signaling/v2/peerconnection.js

```diff
@@ -419,20 +421,18 @@ PeerConnectionV2.prototype._offer = function _offer() {
   }).catch(function() {
     throw new MediaClientLocalDescFailedError();
   }).then(function(offer) {
-    var updatedSdp = self._setCodecPreferences(
-      offer.sdp,
-      self._preferredAudioCodecs,
-      self._preferredVideoCodecs);
-    var updatedOffer = new RTCSessionDescription({
-      type: 'offer',
-      sdp: updatedSdp
-    });
     self._shouldOffer = false;
     if (!self._negotiationRole) {
       self._negotiationRole = 'offerer';
       self._needsInitialAnswer = true;
     }
-    return self._setLocalDescription(updatedOffer);
+    return self._setLocalDescription({
+      type: 'offer',
+      sdp: self._setCodecPreferences(
+        offer.sdp,
+        self._preferredAudioCodecs,
+        self._preferredVideoCodecs)
+    });
```

Comment:
What's the difference here? Is the issue that we called

```js
new RTCSessionDescription
```

instead of

```js
new self._RTCSessionDescription
```

?

Reply:
@markandrus `._setLocalDescription` accepts a Dict. So we should pass the dict directly instead of wrapping it in `self._RTCSessionDescription`.

Reply:
I think anywhere the RTCSessionDescriptionInit is acceptable, the RTCSessionDescription should be acceptable, too.